plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/connection/connection.py

```diff
@@ -35,3 +40,46 @@ def clean_up_fqdn(self, fqdn):
     def test(self):
         self.xdr_api.test_connection()
         return {"success": True}
+
+    def test_task(self):
+        self.logger.error("Running a Task Connection Test for Palo Alto Cortex")
+        return_message = "The connection test to Palo Alto Cortex has failed \n"
+
+        # Gets the current time and converts to  Epoch Unix time (as get_alerts params require date to be an int)
+        now = datetime.now(timezone.utc)
+        start_time = (now - timedelta(minutes=5)).isoformat()
+        start_time = datetime.fromisoformat(start_time)
+        # Timestamp in logs is in Miliseconds
+        start_time = int(start_time.timestamp() * 1000)
+
+        end_time = now.isoformat()
+        end_time = datetime.fromisoformat(end_time)
+        end_time = int(end_time.timestamp() * 1000)
+
+        try:
+            _ = self.xdr_api.get_alerts(start_time, end_time)
```

Comment:
Just noticed this calls get_alerts whereas the task calls get_response_alerts. I think this and the error handling might need changed. Might also be worth checking the messages raised in the existing _post_to_api method need applied to the new make_request within get_response_alerts.

Reply:
Thanks for noticing this